common/persistence/nosql/nosqlplugin/cassandra/workflowUtils.go

```diff
@@ -1205,8 +1203,26 @@ func (db *cdb) updateWorkflowExecution(
 ) error {
 	execution.StartTimestamp = db.convertToCassandraTimestamp(execution.StartTimestamp)
 	execution.LastUpdatedTimestamp = db.convertToCassandraTimestamp(execution.LastUpdatedTimestamp)
-
-	batch.Query(templateUpdateWorkflowExecutionWithVersionHistoriesQuery,
+	//default TTL Value. 0 TTL means no ttl is set, hence your records will persist forever unless explicitly deleted.
+	ttlInSeconds := 0
+	//Only fires when the workflow is closing.
+	if execution.CloseStatus == 1 {
```

Comment:
this condition doesn't cover all closed workflows https://github.com/uber/cadence/blob/714aec48796b62af163e9d324462a1cc2afb4981/common/persistence/dataManagerInterfaces.go#L129C5-L129C5

Reply:
I think you should check `state == WorkflowStateCompleted` instead https://github.com/uber/cadence/blob/714aec48796b62af163e9d324462a1cc2afb4981/common/persistence/dataManagerInterfaces.go#L120